src/main/java/crawlercommons/sitemaps/sax/RSSHandler.java

```diff
@@ -109,47 +107,55 @@ public void startElement(String uri, String localName, String qName, Attributes
      */
     @Override
     public void endElement(String uri, String localName, String qName) throws SAXException {
-        if ("link".equals(currentElement()) || "guid".equals(currentElement())) {
+        if ("link".equals(localName)) {
             setLocURL();
-        } else if ("item".equals(currentElement())) {
+        } else if ("guid".equals(localName)) {
+            // accept as link if
+            // - a valid absolute URL (not a URN, UUID or similar)
+            // - and no <link> found yet
+            if (locURL == null) {
+                setLocURL();
+            }
+            resetCharacterBuffer();
+        } else if ("item".equals(localName)) {
             maybeAddSiteMapUrl();
-        } else if ("rss".equals(currentElement())) {
+        } else if ("rss".equals(localName)) {
```

Comment:
Same question about switch from `currentElement` to `localname`.

Reply:
Calling `currentElement()` returns the content of `localName` because DelegatorHandler pushes `localName` to the element stack before calling the delegate handler. Only inside `characters()` there is no `localName`. I'll make sure that only `localName` is used in all start/endElement methods, if a value is passed as parameter it should be used instead of accessing the element stack.